### ISSUE TITLE
Correlation Feature Selection 

### ISSUE THREAD

**Issue by mmarcato** [fraunhoferportugal/tsfel#91]
Hello there, 

Not an issue, just a suggestion. 

I saw that you created a function in signal_processing called correlated_features which identifies highlighy correlated features given a threshold and then returns a list of features to drop. There is an example on how to use it in the notebook [TSFEL_HAR_Example.ipynb](https://github.com/fraunhoferportugal/tsfel/blob/master/notebooks/TSFEL_HAR_Example.ipynb), but you don't use that in pipeline.

I've used that function as a base to create a class that can be used in pipelines. Please note that I think you should update your function to take the absolute number of the correlation as currently you are only dropping high positively correlated features (with correlation bigger than positive threshold). However, correlations have range [-1,1], so I would have thought you also want to drop high negatively correlated features (i.e. with correlation smaller than negative threshold) . See my code below.

``` 
class CorrelationThreshold(BaseEstimator, TransformerMixin):
    
    """Feature selector that removes all correlated features.

    This feature selection algorithm looks only at the features (X), not the
    desired outputs (y), and can thus be used for unsupervised learning.
    
    Parameters
    ----------
    threshold : float, default=0.95
        Features with a training-set correlation higher than this threshold will
        be removed. The default is to keep all features with non-zero variance,
        i.e. remove the features that have the same value in all samples.

    Attributes
    ----------
    selected_features_ : list, shape (n_features)
        Returns a list with the selected feature names.

    """

    def __init__(self, threshold = 0.95):
        self.threshold = threshold
        self.to_drop = None
        self.to_keep = None

    def fit (self, X, y = None ): 
        '''
        Parameters
        ----------
        X : {array-like, sparse matrix}, shape (n_samples, n_features)
            Sample vectors from which to compute variances.
        y : any, default=None
            Ignored. This parameter exists only for compatibility with
            sklearn.pipeline.Pipeline.
        Returns
        -------
        self
        '''
        corr_matrix = np.absolute(X.corr())
        upper = corr_matrix.where(np.triu(np.ones(corr_matrix.shape), k=1).astype(np.bool))
        self.to_drop = [column for column in upper.columns if any(upper[column] > self.threshold)]
        self.to_keep = list(set(X.columns) - set(self.to_drop))
        return self
        
    def transform(self, X, y = None):
        X_selected = X[self.to_keep]
        return X_selected
    
    def get_support(self):
        return self.to_keep

```

Please note that this is a topic in discussion in the sklearn community too: [#13405](https://github.com/scikit-learn/scikit-learn/issues/13405), [#14698](https://github.com/scikit-learn/scikit-learn/pull/14698). One thing that should be further discussed/analysed/improved is how to choose which of the highly correlated variables should be dropped. I think some , f-value correlation with the y label could be appropriate - only applicable to supervised learning problems.

**Comment by dmfolgado** [comment]
Hi! Thanks for your suggestion. We will take into account your considerations. Regarding the negative correlation, it was a known behavior, but I agree that it also must be taken into account more explicitly.

**Comment by dmfolgado** [comment]
Hi! Thanks for your suggestion. We will take into account your considerations. Regarding the negative correlation, it was a known behavior, but I agree that it also must be taken into account more explicitly.

**Comment by mmarcato** [comment]
Hello, 
Thanks for your reply ;D Yes, about the negative correlation, it is not removing those variables at all the way it is set up now.

**Comment by dmfolgado** [comment]
We committed some changes today on the development branch.
The absolute value of correlation is taken into account from now on to remove correlated features.
